react/src/views/Transfers/CreateTransferAgreement/components/CreateTransferAgreement.tsx

```diff
@@ -0,0 +1,269 @@
+import {
+  Box,
+  Button,
+  ButtonGroup,
+  FormLabel,
+  Heading,
+  Input,
+  List,
+  ListItem,
+  Stack,
+} from "@chakra-ui/react";
+import { useEffect, useState } from "react";
+import { SubmitHandler, useForm } from "react-hook-form";
+import { zodResolver } from "@hookform/resolvers/zod";
+import { z } from "zod";
+import SelectField, { IDropdownOption } from "components/Form/SelectField";
+import RadioGroupField from "components/Form/RadioGroupField";
+import DateField from "components/Form/DateField";
+
+export interface IBaseData {
+  id: string;
+  name: string;
+}
+
+export interface IBasesForOrganisationData {
+  id: string;
+  name: string;
+  bases: IBaseData[];
+}
+
+const singleSelectOptionSchema = z.object({
+  label: z.string(),
+  value: z.string(),
+});
+
+export const TransferAgreementFormDataSchema = z.object({
+  currentOrganisationSelectedBases: singleSelectOptionSchema
+    .array()
+    .min(1)
+    .nonempty("Please select at least one base"),
+  transferType: z.string().nullable().refine(Boolean, { message: "Please choose a transfer type" }),
+
+  partnerOrganisation: singleSelectOptionSchema
+    .refine(Boolean, { message: "Please select a partner organisation" })
+    .transform((selectedOption) => selectedOption || { label: "", value: "" }),
+  partnerOrganisationSelectedBases: singleSelectOptionSchema.array().optional(),
+  validFrom: z
+    .date({
+      required_error: "Please enter a valid date",
+      invalid_type_error: "Please enter a valid date",
+    })
+    .optional()
+    .transform((value) => value?.toISOString().substring(0, 10)),
+  validUntil: z
+    .date()
+    .optional()
+    .transform((value) => value?.toISOString().substring(0, 10)),
+  comment: z.string().optional(),
+});
+// .refine(
+//   (data) => {
+//     // eslint-disable-next-line no-console
+//     console.log(data);
+//     if (typeof data.validFrom !== "undefined" && typeof data.validUntil !== "undefined") {
+//       const dateDiff = data.validUntil.getTime() - data.validFrom.getTime();
+//       return dateDiff > 0;
+//     }
+//     return true;
+//   },
+//   {
+//     message: "Please enter greater date for the valid until",
+//     path: ["validUntil"], // path of error
+//   },
+// );
```

Comment:
Comment can probably removed?

Reply:
Validating that the second date is later than the first should be done in the back-end, too.
If you want, you can add an item to check it in the FE to the "General enhancements" trello card.